datacommons/stat_vars.py

```diff
@@ -203,17 +203,33 @@ def get_stat_all(places, stat_vars):
       }
     """
     url = utils._API_ROOT + utils._API_ENDPOINTS['get_stat_all']
-    req_json = {'stat_vars': stat_vars, 'places': places}
-
-    # Send the request
-    res_json = utils._send_request(url, req_json=req_json, use_payload=False)
-
-    if 'placeData' not in res_json:
-        raise ValueError('No data in response.')
-
-    # Unnest the REST response for keys that have single-element values.
-    place_statvar_series = collections.defaultdict(dict)
-    for place_dcid, place in res_json['placeData'].items():
-        for stat_var_dcid, stat_var in place['statVarData'].items():
-            place_statvar_series[place_dcid][stat_var_dcid] = stat_var
-    return dict(place_statvar_series)
+    places = list(places)
+    # Ceil hack to get # of batches.
+    batches = -(-len(places) // utils._QUERY_BATCH_SIZE)
+    res = {}
+    no_data = True
+    for i in range(batches):
+        req_json = {
+            'stat_vars':
+                stat_vars,
+            'places':
+                places[i * utils._QUERY_BATCH_SIZE:(i + 1) *
+                       utils._QUERY_BATCH_SIZE]
+        }
+        # Send the request
+        res_json = utils._send_request(url,
+                                       req_json=req_json,
+                                       use_payload=False)
+        if 'placeData' in res_json:
+            no_data = False
+
+        # Unnest the REST response for keys that have single-element values.
+        place_statvar_series = collections.defaultdict(dict)
+        for place_dcid, place in res_json['placeData'].items():
+            for stat_var_dcid, stat_var in place['statVarData'].items():
+                place_statvar_series[place_dcid][stat_var_dcid] = stat_var
+        res.update(dict(place_statvar_series))
+
+    if no_data:
+        raise ValueError('No data in responses.')
```

Comment:
This is interesting - so we throw an error even if one out of many batches fail. Is that the right response (perhaps we can add more info in the error)? We should also throw earlier instead of doing this outside the loop.

Reply:
Also thinking about the behavior, shall we just return empty result for the entry with no data in stead of throwing error? For example, if someone get stats all for crime for city, county, state, it seems a bit intrusive to throw error because we don't have data for county?

Reply:
I think it's actually only returning no_data if placeData was not in any of the responses. no_data is initally set to True, then each time placeData exists (the most top level key of REST response), then no_data is set to False. It is never set to True again. I could optimize it s.t. we only assign False if it is currently True, but the logic is the same. This is also why it is outside the loop--so that as long as one batch has data, we return a response.

I will add an else statement that continues the for loop so that we don't have a key error. Will also add a test for this.